fast/stages/2-security/vpc-sc.tf

```diff
@@ -19,38 +19,117 @@ locals {
+    for k in local._vpc_sc_resources_lookup_folders :
+    k => formatlist("projects/%s", data.google_projects.project_in_folders[k].projects[*].number)
+  }
+  _vpc_sc_resources_projects_by_ids = {
+    for k in flatten(values(data.google_projects.project_by_ids)[*].projects) :
+    k.project_id => "projects/${k.number}"
+  }
   # compute the number of projects in each perimeter to detect which to create
   vpc_sc_counts = {
     for k, v in var.vpc_sc_perimeters : k => length(v.resources)
   }
   # define dry run spec at file level for convenience
   vpc_sc_explicit_dry_run_spec = true
   # compute perimeter bridge resources (projects)
-  vpc_sc_bridge_resources = {
-    landing_to_dev = concat(
-      var.vpc_sc_perimeters.landing.resources,
-      var.vpc_sc_perimeters.dev.resources
-    )
-    landing_to_prod = concat(
-      var.vpc_sc_perimeters.landing.resources,
-      var.vpc_sc_perimeters.prod.resources
-    )
+  vpc_sc_perimeter_resources = {
+    for k, v in var.vpc_sc_perimeters : k => sort(toset(flatten([
+      v.resources,
+      [for f in v.folder_ids : local._vpc_sc_resources_projects_in_folders[f]],
+      [for f in v.project_ids : local._vpc_sc_resources_projects_by_ids[f]]
+    ])))
   }
-  # compute spec/status for each perimeter
   vpc_sc_perimeters_spec_status = {
-    dev = merge(var.vpc_sc_perimeters.dev, {
-      restricted_services     = local._vpc_sc_restricted_services
-      vpc_accessible_services = local._vpc_sc_vpc_accessible_services
-    })
-    landing = merge(var.vpc_sc_perimeters.landing, {
-      restricted_services     = local._vpc_sc_restricted_services
-      vpc_accessible_services = local._vpc_sc_vpc_accessible_services
-    })
-    prod = merge(var.vpc_sc_perimeters.prod, {
-      restricted_services     = local._vpc_sc_restricted_services
-      vpc_accessible_services = local._vpc_sc_vpc_accessible_services
-    })
+    for k, v in var.vpc_sc_perimeters : k => merge([
+      v,
+      {
+        resources               = local.vpc_sc_perimeter_resources[k]
+        restricted_services     = local._vpc_sc_restricted_services
+        vpc_accessible_services = local._vpc_sc_vpc_accessible_services
+        project_ids             = null
+        folder_ids              = null
+      },
+    ]...)
+  }
+  vpc_sc_bridge_resources = {
+    for p in setproduct(["landing"], ["dev", "prod"]) :
+    "${p.0}_to_${p.1}" => sort(toset(flatten([
+      local.vpc_sc_perimeter_resources[p.0],
+      local.vpc_sc_perimeter_resources[p.1],
+    ])))
+  }
+  vpc_sc_egress_policies = {
+    for k, v in var.vpc_sc_egress_policies :
+    k => {
+      from = v.from
+      to = merge([
+        v.to,
+        {
+          resources = sort(toset(flatten([
+            v.to.resources,
+            [for f in v.to.folder_ids : local._vpc_sc_resources_projects_in_folders[f]],
+            [for f in v.to.project_ids : local._vpc_sc_resources_projects_by_ids[f]],
+          ])))
+          project_ids = null
```

Comment:
I think this is ignored by the [provider](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/access_context_manager_service_perimeter). Looks lit it only accepts resources in the form of projects/number

Reply:
Yes, I nulled it in case of underlying module changes